File: Two-Sum-Problem/twoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> valueMap = new HashMap<>(); // value : index
        for (int i = 0; i < nums.length; ++i) {
            int diff = target - nums[i];
            if (valueMap.containsKey(diff)) {   // if difference found
                return new int[] { valueMap.get(diff), i};
            }
            valueMap.put(nums[i], i);   // if difference not found
        }
        return null; // no solution
    }
}

File: Two-Sum-Problem/twoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> valueMap; // value : index
        vector<int> output;
        for (int i = 0; i < nums.size(); ++i) {
            int diff = target - nums[i];
            if (valueMap.find(diff) == valueMap.end()) {    // difference not found
                valueMap.insert(pair<int, int>(nums[i], i));
            } else {    // difference found
                output.push_back(valueMap[diff]);
                output.push_back(i);
                return output;
            }
        }
        return output;  // no solution
    }
};